index.bs

```diff
@@ -6145,6 +6244,354 @@ However, [=authenticators=] that do not utilize [[!FIDO-CTAP]] do not necessaril
+        ArrayBuffer authenticatorOutput;
+        ArrayBuffer signature;
+    };
+
+    partial dictionary AuthenticationExtensionsClientOutputs {
+        AuthenticationExtensionsDevicePublicKeyOutputs devicePubKey;
+    };
+    </xmp>
+
+
+: Authenticator extension input
+:: A CBOR expression of the client extension input
+
+    ```
+    devicePublicKeyInputs = {
+      attestation: "none" / "indirect" / "direct" / "enterprise",
+      attestationFormats: [tstr],
+    }
+    $$extensionInput //= (
+      devicePubKey: devicePublicKeyInputs,
+    )
+    ```
+
+: Authenticator extension output
+:: The <dfn>device public key attestation object</dfn>, defined by the `attObjForDevicePublicKey` type:
+
+    ```
+    $$extensionOutput //= (
+        devicePubKey: attObjForDevicePublicKey,
+    )
+
+    attObjForDevicePublicKey = { ; Note: This object conveys an attested
+                                 ; device public key and is analogous to \`attObj\`.
+
+        aaguid:  bstr,  ; Authenticator's AAGUID (16 bytes fixed-length)
+                        ; https://www.w3.org/TR/webauthn/#aaguid
+
+        dpk:     bstr,  ; The Device Public Key (self-describing variable length,
+                        ; COSE_Key format, CBOR-encoded)).
+
+        ; Whether this key is scoped to the entire device, or a loosely-defined,
+        ; narrower scope called "app". For example, a "device"-scoped key is expected
+        ; to be the same between an app and a browser on the same device, while
+        ; an "app"-scoped key would probably not be.
+        ;
+        ; Whatever the scope, a device key is still specific to a given credential
+        ; and does not provide any ability to link credentials.
+        ;
+        ; Whether device-scoped or not, keys are still device-bound. I.e. an
+        ; app-scoped key does not enjoy lesser protection from extraction.
+
+        scope: uint .size 1, ; a value of 0x00 means "entire device" ("all apps") scope.
+                             ; 0x01 means "per-app" scope.
+                             ; Values other than 0x00 or 0x01 are reserved for future
+                             ; use.
+
+        ; An authenticator-generated random nonce for inclusion in the attestation
+        ; signature. If the authenticator chooses to not generate a nonce, it sets this
+        ; to a zero-length bytestring.
+
+        nonce: bstr .size (0..32),
+
+        ; See https://www.w3.org/TR/webauthn/#sctn-generating-an-attestation-object
+        ;
+        ; Attestation statement formats define the \`fmt\` and \`attStmt\` members of
+        ; $$attStmtType.
+        ;
+        ; In summary, the \`attStmt\` will (typically) contain:
+        ;   (1) a SIGNATURE value calculated (using the attestation private key)
+        ;       over (prefix || aaguid || dpk || nonce) where \`prefix\` is
+        ;       h'64657669636520626f756e64206b6579206174746573746174696f6e20736967
+        ;         00ffffffff'
```

Comment:
What is this prefix value all about? Where does it come from, and what is its purpose?

Reply:
the prefix translates to (hex-decoding): "device bound key attestation sig" followed by 00ffffffff.
It explains what was signed and ensures that the to-be-signed object can never be confused with some other to-be-signed object

Reply:
> What is this prefix value all about? Where does it come from, and what is its purpose?

Similar to the nonce, I see the purpose is explained later in the spec, and I like many was reading from top-down. Is it possible to reference the purpose from here?